### PR TITLE
Core/Spells: Fix display for Shaman totems

### PR DESCRIPTION
**Pull request by DoctorKraft**
**Changes proposed:**

-  Store models of totems in database
-  Totems have now the correct displayid

**Target branch(es):**

- [X] master

**Issues addressed:**

Closes #Unk

**Tests performed:**

Tested IG & Build

**Known issues and TODO list:** (add/remove lines as needed)

- [x] More SQL stuff

### CODE REVIEW

File: src/server/game/Spells/SpellMgr.h [hunk 1]
```diff
@@ -685,6 +693,8 @@ class TC_GAME_API SpellMgr
 
         void LoadPetFamilySpellsStore();
 
+        uint32 GetModelForTotem(uint32 spellID, uint32 radeID) const;
```
Comment (DoctorKraft) [inline src/server/game/Spells/SpellMgr.h]: Note: Fix that

File: sql/updates/world/master/9999_99_99_99_world.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
+--
+-- Table structure for table `spell_totem_model`
+--
+DROP TABLE IF EXISTS `spell_totem_model`;
+CREATE TABLE `spell_totem_model` (
+  `SpellID` mediumint(8) unsigned NOT NULL,
+  `RaceID` mediumint(8) unsigned NOT NULL,
+  `DisplayID` mediumint(8) unsigned NOT NULL DEFAULT '0',
+  PRIMARY KEY (`SpellID`,`RaceID`,`DisplayID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```
Comment (mdX7) [inline sql/updates/world/master/9999_99_99_99_world.sql]: MyISAM instead of InnoDB

File: src/server/game/Entities/Totem/Totem.cpp [hunk 3]
```diff
@@ -76,7 +76,10 @@ void Totem::InitStats(uint32 duration)
             });
 
             if (summonEffect != effects.end())
-                SetDisplayId(owner->GetModelForTotem(PlayerTotemType((*summonEffect)->MiscValueB)));
+            {
+                uint32 totemDisplayID = sSpellMgr->GetModelForTotem(createdBySpell->Id, owner->getRace());
+                SetDisplayId(totemDisplayID);
```
Comment (mdX7) [inline src/server/game/Entities/Totem/Totem.cpp]: when `sSpellMgr->GetModelForTotem` returns 0 the DisplayId shouldn't be set

File: src/server/game/Spells/SpellMgr.h [hunk 4]
```diff
@@ -593,6 +593,14 @@ struct SpellInfoLoadHelper
     SpellTotemsEntry const* Totems = nullptr;
 };
 
+struct TC_GAME_API SpellTotemModel
+{
+    uint32 raceID;
```
Comment (mdX7) [inline src/server/game/Spells/SpellMgr.h]: iirc we are using UpperCamelCase for structs, not sure though, kinda mixed in current tc code

**Review by Shauren — APPROVED** [review]

**Pull request merged**